### PR TITLE
Tegra downstream rebase 1.3.19

### PR DESCRIPTION
**Pull request by vwadekar**
Rebase downstream Tegra patches on TF-A v2.0.

Compilation works for all supported platforms: T132, T210, T186

### CODE REVIEW

File: plat/nvidia/tegra/common/drivers/bpmp/bpmp.c [hunk 1]
```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/common/drivers/bpmp/bpmp.c]: Use SPDX identifier. Also, for new files you can simply put Nvidia as copyright holder.

File: plat/nvidia/tegra/include/drivers/bpmp.h [hunk 2]
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/include/drivers/bpmp.h]: Same here.

File: plat/nvidia/tegra/include/drivers/security_engine.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
+﻿/*
+ * Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/include/drivers/security_engine.h]: SPDX identifier.

File: plat/nvidia/tegra/include/drivers/security_engine.h [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * and proprietary rights in and to this software, related documentation
+ * and any modifications thereto.  Any use, reproduction, disclosure or
+ * distribution of this software and related documentation without an express
+ * license agreement from NVIDIA CORPORATION is strictly prohibited.
+ */
+
+#ifndef __SECURITY_ENGINE_H__
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/include/drivers/security_engine.h]: Using double underscore is disallowed in MISRA. Use ``SECURITY_ENGINE_H``.

File: plat/nvidia/tegra/soc/t210/drivers/se/se_private.h [hunk 5]
```diff
@@ -0,0 +1,229 @@
+﻿/*
+ * Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/soc/t210/drivers/se/se_private.h]: SPDX.

File: plat/nvidia/tegra/soc/t210/drivers/se/security_engine.c [hunk 6]
```diff
@@ -0,0 +1,413 @@
+﻿/*
+ * Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/soc/t210/drivers/se/security_engine.c]: SPDX

**Comment by vwadekar** [comment]
Fixed review comments and pushed updated patches

File: plat/nvidia/tegra/include/drivers/security_engine.h [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+/*
+ * Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
+ * and proprietary rights in and to this software, related documentation
+ * and any modifications thereto.  Any use, reproduction, disclosure or
+ * distribution of this software and related documentation without an express
+ * license agreement from NVIDIA CORPORATION is strictly prohibited.
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/include/drivers/security_engine.h]: This header is still problematic:
> Any use, reproduction, disclosure or distribution of this software and related documentation without an express license agreement from NVIDIA CORPORATION is strictly prohibited.

File: plat/nvidia/tegra/soc/t210/drivers/se/se_private.h [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+ */
+
+/*
+ * Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
+ * and proprietary rights in and to this software, related documentation
+ * and any modifications thereto.  Any use, reproduction, disclosure or
+ * distribution of this software and related documentation without an express
+ * license agreement from NVIDIA CORPORATION is strictly prohibited.
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/soc/t210/drivers/se/se_private.h]: Same, this header is problematic.

**Comment by vwadekar** [comment]
Fixed the problematic headers

File: plat/nvidia/tegra/soc/t210/drivers/se/security_engine.c [hunk 9]
```diff
@@ -0,0 +1,512 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
+ * and proprietary rights in and to this software, related documentation
+ * and any modifications thereto.  Any use, reproduction, disclosure or
+ * distribution of this software and related documentation without an express
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/soc/t210/drivers/se/security_engine.c]: This header is still here.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
You have to fix a ```checkpatch.pl``` error:
```
plat/nvidia/tegra/soc/t186/plat_psci_handlers.c:286: ERROR: space prohibited before that ',' (ctx:WxW)
+		ERROR("%s: unsupported CPU (0x%lx)\n", __func__ , mpidr);
```

**Comment by antonio-nino-diaz-arm** [comment]
I normally do:
```
CHECKPATCH=../linux/scripts/checkpatch.pl make checkpatch
```
If you are in a branch this will check all patches between HEAD and the first common patch between your branch and master.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Pull request merged**